### ISSUE TITLE
Allow multiple 'it' references in UnnecessaryLet

### ISSUE THREAD

**Issue by Nycto** [arturbosch/detekt#1359]
## Expected Behavior
I expect this code to be considered valid by the `UnnecessaryLet` validator:

```kotlin
fun f() {
    val a : Int? = null
    a?.let { it.plus(it) }
    a?.let { foo -> foo.plus(foo) }
}
```

## Observed Behavior
The `UnnecessaryLet` validator reports a code smell, even though the parameter can't be removed -- it's referenced multiple times

## Steps to Reproduce
The code above is easy to drop into a test to confirm this is broken

## Context
This creates false positives that require a `Suppress` statement to work around

## Your Environment
* Version of detekt used: 565d86d780c15ade77b5468a9e342ba334181333
* Version of Gradle used (if applicable): 4.10.2
* Operating System and version: Linux 4.4.0-93-generic amd64
* Link to your project (if it's a public repository): Private repo
